Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -1103,12 +1106,44 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, CoalesceObjectBreak
     return { };
 }
 
+bool AXTextMarker::shouldIncrementOffsetForAmbiguousPosition() const
+{
+    if (!isDownstream()) {
+        // This disambiguation is only required for downstream positions, as they are more "forward"
+        // than upstream positions. Incrementing the downstream position moves us to an unambiguous
+        // position.
+        return false;
+    }
+    // This function should only be called on a text marker we know to be a valid text run marker.
+    ASSERT(isValid() && isInTextRun());
+    const auto* runs = this->runs();
+    size_t runIndex = runs->indexForOffset(offset());
+
+    size_t firstOffsetInRun = runIndex ? runs->runLengthSumTo(runIndex - 1) : 0;
+    size_t lastOffsetInRun = runs->runLengthSumTo(runIndex);
+    if (offset() == firstOffsetInRun) {
+        // If we are at the first offset in a run that isn't the first one, this position is ambiguous.
+        // Example, where _ is a whitespace:
+        // foo_
+        // |bar
+        return runIndex;
```

Comment:
Optional, but I'd favor return runIndex > 0; so it's clear you're returning a bool

Reply:
Agreed, fixed!